strapi/strapi
# [SSO] Accept null value for ssoLockedRoles

### What does it do?

Allow the backend validation to accept a null value for ssoLockedRoles.

### Why is it needed?

Fixes a bug where the Settings -> SSO form will not save unless a role is first selected and deselected (allowing the front-end to submit an empty array rather than null)

The frontend receives a null as the initial value and if the field is unchanged it will send it back, which was causing the problem.

### How to test it?

Starting from a new project (or one where where the Strapi option for providers.ssoLockedRoles is null) attempt to save the Settings -> SSO form with an empty SSO locked roles multiselect.

It should now work and accept the null value.

API tests have been added for provider options. Reminder that they only run with EE enabled, that is with `STRAPI_LICENSE=your_license yarn strapi test:api`

### Related issue(s)/PR(s)

Fixes #17172
